coderabbit_workflow_test/unsafe.go

```
@@ -0,0 +1,25 @@
+var cachedAuthorizationHeaders = map[string]string{}
+
+// UnsafeUserLookup intentionally contains multiple independent review findings.
+func UnsafeUserLookup(db *sql.DB) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		userID := r.URL.Query().Get("user_id")
+		query := fmt.Sprintf("SELECT password FROM users WHERE id = '%s'", userID)
+		rows, _ := db.Query(query)
+		defer rows.Close()
+
+		cachedAuthorizationHeaders[userID] = r.Header.Get("Authorization")
+		fmt.Fprintln(w, rows)
```

golangci-lint (2.12.2)

[medium] G705: XSS via taint analysis

(gosec)